lib/Conch/Route/Device.pm

```diff
@@ -14,128 +14,496 @@ Conch::Route::Device
-    POST    /device/:device_id/location
-    DELETE  /device/:device_id/location
-
-    GET     /device/:device_id/settings
-    POST    /device/:device_id/settings
-    GET     /device/:device_id/settings/#key
-    POST    /device/:device_id/settings/#key
-    DELETE  /device/:device_id/settings/#key
-
-    POST    /device/:device_id/validation/:validation_id
-    POST    /device/:device_id/validation_plan/:validation_plan_id
-    GET     /device/:device_id/validation_state
-
-    GET     /device/:device_id/interface
-    GET     /device/:device_id/interface/:interface_name
-    GET     /device/:device_id/interface/:interface_name/:field
-
 =cut
 
 sub routes {
     my $class = shift;
     my $device = shift; # secured, under /device
 
+=head3 C<POST /device/:device_id>
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<ValidationStateWithResults|https://conch.joyent.us/schema/response/ValidationStateWithResults>
+
+=back
+
+=cut
+
     # POST /device/:device_id
     $device->post('/:device_id')->to('device_report#process');
 
+=head3 C<GET /device?key=:value>
+
+Looks up one or more devices by query parameter. Supports:
+
+    /device?hostname=$hostname
+    /device?mac=$macaddr
+    /device?ipaddr=$ipaddr
+    /device?$setting_key=$setting_value
+
+Returns a C<403 FORBIDDEN> if the current user doesn't have permissions on the
+device through at least one workspace.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<Devices|https://conch.joyent.us/schema/response/Devices>
+
+=back
+
+=cut
+
     # GET /device?key=:value
     $device->get('/')->to('device#lookup_by_other_attribute');
 
     {
         # chainable action that extracts and looks up device_id from the path
         my $with_device = $device->under('/:device_id')->to('device#find_device');
 
+=head3 C<GET /device/:device_id>
+
+Retrieves details about a single (active) device.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<DetailedDevice|https://conch.joyent.us/schema/response/DetailedDevice>
+
+=back
+
+=cut
+
         # GET /device/:device_id
         $with_device->get('/')->to('device#get');
 
+=head3 C<GET /device/:device_id/pxe>
+
+Gets PXE-specific information about a device.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<DevicePXE|https://conch.joyent.us/schema/response/DevicePXE>
+
+=back
+
+=cut
+
         # GET /device/:device_id/pxe
         $with_device->get('/pxe')->to('device#get_pxe');
 
+=head3 C<GET /device/:device_id/phase>
+
+Gets just the device's phase.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<DevicePhase|https://conch.joyent.us/schema/response/DevicePhase>
+
+=back
+
+=cut
+
         # GET /device/:device_id/phase
         $with_device->get('/phase')->to('device#get_phase');
 
+=head3 C<POST /device/:device_id/graduate>
+
+Marks the device as "graduated" (VLAN flipped). Returns C<409 CONFLICT> if the
```

Comment:
This description of 'graduated' is not accurate

Reply:
I've removed the description entirely until we better decide how to document the relationships between endpoints.